packages/react-theming/src/components/Button/BaseButton.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+
+/**
+ * TODO:
+ * 1) do we really need slots prop?
```

Comment:
Do you mean keeping this in context only opposed to also having a slot?

```jsx
<Provider theme={{
    components: {
      Button: {
        slots: { icon: MyIcon },
      }
    }
  }}
/>
```
>Ignore the fact that I put this in `theme.components`. I just mean to show "top-level config via context" vs a prop.


vs

```jsx
<Button slots={{ icon: MyIcon }} />
```

This would allow easy inline overrides of the slots, but is it necessary?